core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,161 @@
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	blockSubscriber    *evm21.BlockSubscriber
+	blockSubChanID     int
+	threadCtrl         utils.ThreadControl
+	lggr               logger.Logger
+	configDigest       [32]byte
+	latestConfigDigest latestConfigDigestGetter
+}
+
+type latestConfigDigestGetter interface {
+	LatestConfigDetails(opts *bind.CallOpts) (keeper_registry_wrapper2_0.LatestConfigDetails, error)
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint,
+	lggr logger.Logger, chain evm.Chain, rAddr common.Address, blocksub *evm21.BlockSubscriber) (*AutomationCustomTelemetryService, error) {
+	registry, rErr := keeper_registry_wrapper2_0.NewKeeperRegistry(rAddr, chain.Client())
+	if rErr != nil {
+		return nil, errors.Wrap(rErr, "error creating new Registry Wrapper for customTelemService")
+	}
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint: me,
+		threadCtrl:         utils.NewThreadControl(),
+		lggr:               lggr.Named("Automation Custom Telem"),
+		latestConfigDigest: registry,
+		blockSubscriber:    blocksub,
+	}, nil
+}
+
+// Start starts Custom Telemetry Service, sends 1 NodeVersion message to endpoint at start and sends new BlockNumber messages
+func (e *AutomationCustomTelemetryService) Start(ctx context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		e.lggr.Infof("Starting: Custom Telemetry Service")
+		callOpt := &bind.CallOpts{Context: ctx}
+		configDetails, cdErr0 := e.latestConfigDigest.LatestConfigDetails(callOpt)
+		if cdErr0 != nil {
+			e.lggr.Errorf("Error occurred while getting newestConfigDetails for initialization %v", cdErr0)
+		} else {
+			e.configDigest = configDetails.ConfigDigest
+			e.sendNodeVersionMsg()
+		}
+		e.threadCtrl.Go(func(ctx context.Context) {
+			ticker := time.NewTicker(1 * time.Minute)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					callOpt := &bind.CallOpts{Context: ctx}
+					newConfigDetails, cdErr := e.latestConfigDigest.LatestConfigDetails(callOpt)
+					if cdErr != nil {
+						e.lggr.Errorf("Error occurred while getting newestConfigDetails  %v", cdErr)
+						continue
+					}
+					newConfigDigest := newConfigDetails.ConfigDigest
+					if newConfigDigest != e.configDigest {
+						e.configDigest = newConfigDigest
+						e.sendNodeVersionMsg()
+					}
+				case <-ctx.Done():
+					return
+				}
+			}
+		})
+
+		chanID, blockSubscriberChan, blockSubErr := e.blockSubscriber.Subscribe()
+		if blockSubErr != nil {
+			e.lggr.Errorf("Block Subscriber Error: Subscribe(): %s", blockSubErr)
+		}
+		e.blockSubChanID = chanID
+		e.threadCtrl.Go(func(ctx context.Context) {
+			e.lggr.Infof("Started: Sending BlockNumber Messages")
+			for {
+				select {
+				case blockHistory := <-blockSubscriberChan:
```

Comment:
what would happen if processing of block history takes longer than the next block being sent in channel? I think it will be dropped but would be good to double check